packages/compartment-mapper/src/link.js

```diff
@@ -372,16 +411,42 @@ export const link = (
     // The `moduleMapHook` writes back to the compartment map.
     compartmentDescriptor.modules = modules;
 
+    /** @type {Record<string, Language>} */
+    const finalLanguageForExtension = Object.create(null);
+    for (const [extension, language] of entries(customLanguageForExtension)) {
+      if (extension in languageForExtension) {
```

Comment:
Should I be using `has` instead of this?  It's habit (because TypeScript seems to prefer it).

Reply:
Since they’re equivalent for `Object.create(null)`, I’m inclined to overlook the question. Unless I’m wrong about them being equivalent. @erights for second opinion.

Reply:
I did some additional work to ensure that `in` behaves as I would expect. In the case of `options` where a devious user might provide an object with a prototype chain that includes enumerable properties, the difference between `has` and `in` would be observable, but that can be avoided by capturing options with `freeze(create(null), option)`.